src/status_im2/contexts/activity_center/events.cljs

```diff
@@ -153,6 +157,76 @@
   [cofx notification]
   (notifications-reconcile cofx [(assoc notification :read true)]))
 
+(defn- update-read-flag
+  [notifications read?]
+  (reduce (fn [acc notification]
+            (conj acc (assoc notification :read read?)))
+          []
+          notifications))
+
+(rf/defn mark-all-as-read
+  {:events [:activity-center.notifications/mark-all-as-read]}
+  [{:keys [db] :as cofx}]
+  (when-let [undoable-till (get-in db [:activity-center :mark-all-as-read-undoable-till])]
+    (when (>= (datetime/timestamp) undoable-till)
+      {:db            (update-in db [:activity-center] dissoc :mark-all-as-read-undoable-till)
+       :json-rpc/call [{:method     "wakuext_markAllActivityCenterNotificationsRead"
+                        :params     []
+                        :on-success #(rf/dispatch
+                                      [:activity-center.notifications/mark-all-as-read-success])
+                        :on-error   #(rf/dispatch [:activity-center/process-notification-failure
+                                                   nil
+                                                   :notification/mark-all-as-read
+                                                   %])}]})))
+
+(rf/defn mark-all-as-read-success
+  {:events [:activity-center.notifications/mark-all-as-read-success]}
+  [{:keys [db]}]
+  {:db (update-in db [:activity-center] dissoc :mark-all-as-read-undoable-till)})
+
+(rf/defn undo-mark-all-as-read
+  {:events [:activity-center.notifications/undo-mark-all-as-read-locally]}
+  [{:keys [db] :as cofx} {:keys [unread-count notifications]}]
+  {:db (-> db
+           (update-in [:activity-center :notifications]
+                      update-notifications
+                      notifications)
+           (assoc-in [:activity-center :unread-count] unread-count)
```

Comment:
Unfortunately this must change now @smohamedjavid. The `[:activity-center :unread-count]` state is no longer usable, it's now `[:activity-center :unread-counts-by-type]`. That's because every tab needs a separate count of the unread notifications. You can play around with the app and you should be able to reproduce that the unread indicator in any tab is not updated correctly after marking all notifications as read, nor the undo works as expected.

The new state is a map of *notification type* => *unread count*. See the unit test https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/activity_center/events_test.cljs#L709

Behind the scenes, the subscription `:activity-center/unread-count` sums all the counts to give a total to the UI. See https://github.com/status-im/status-mobile/blob/develop/src/status_im2/subs/activity_center.cljs#L30

Reply:
Apologies, I missed out the `unread-count-by-type`.  During the initial implementation, it was not added into the `develop`. 

No worries, I updated the PR with the updation of unread count by each type upon marking all notifications as read.

Thanks for letting me know, Icaro!